### ISSUE TITLE
Adding links opening in new tab inside ordered or unordered lists puts entire list text a tag aria-label

### ISSUE THREAD

**Issue by yingles** [WordPress/gutenberg#12446]
**Describe the bug**
When adding a link opening in a new tab within an ordered or unordered list the entire text of the list at the time the link is made to open in a new window is output in the aria-label.

**To Reproduce**
1. Add a New Post
2. Create an ordered or unordered list block
3. Type "Link 1"
4. Select text and make it a link to https://example.com opening in a new tab
5. Press Enter to start a new line item and type "Link 2"
6. Select text and make it a link to https://example.com opening in a new tab
7. Press Enter to start a new line item and type "Link 3" (don't link this yet)
8. Press Enter to start a new line item and type "Additional Text"
9. Make "Link 3" a link to https://example.com opening in a new tab

**Expected behavior**
* Link 1 should have `aria-label="Link 1 (opens in a new tab)"`
* Link 2 should have `aria-label="Link 2 (opens in a new tab)"`
* Link 3 should have `aria-label="Link 3 (opens in a new tab)"`

**Actual behavior**
* Link 1 has `aria-label="Link 1 (opens in a new tab)"`
* Link 2 has `aria-label="Link 1Link2 (opens in a new tab)"`
* Link 3 has `aria-label="Link 1Link2Link3Additional Text (opens in a new tab)"`

**Additional notes**
* Going back to Link 1, removing the link, and readding a link to open in a new tab causes Link one to update to a`aria-label="Link 1Link2Link3Additional Text  (opens in a new tab)"`
* Copying and pasting the list removes all aria-labels 

**Desktop (please complete the following information):**
 - OS: macOS 10.14.1
 - Browser Chrome 
 - Version 70.0.3538.102

**Additional context**
- WordPress 4.9.8 fresh install
- Gutenberg plugin 4.5.1
- twentyseventeen
- No additional plugins

**Comment by yingles** [comment]
One other note: I noticed this behavior because it also sucked in some inline shortcodes used within the list and ran them.

**Comment by youknowriad** [comment]
This is a duplicate of #12325
